identity-account/src/identity/identity_state.rs

```diff
@@ -81,6 +81,10 @@ impl IdentityState {
     &mut self.document
   }
 
+  pub fn set_document(&mut self, document: IotaDocument) {
+    self.document = document;
```

Comment:
Discussion: Do we really want trivial setters?
Hiding the mutation of a field inside a function can be good for future proofing, but introduces some indirection when for those reading/auditing code (one has to check that the `set_document` function does not have any other side effects or does other non-trivial things in addition to checking the field). Hence if one is sure about the internal presentation it is perhaps better to just declare that field public.  

Note also that in this case we already have `document_mut` and then one can just compose that with [mem::replace](https://doc.rust-lang.org/std/mem/fn.replace.html) to obtain the same functionality as this setter, so there is even less reason to introduce an extra method here.

Reply:
Personally, I don't mind having this setter. I find that having to use `document_mut` + `mem::replace` would be a bit annoying.

Reply:
As a compromise, I would be perfectly fine with making such fields `pub(crate)`, but not `pub`. With the latter, everything is even more likely to be a breaking change, and we retain less control of the code. Also note that `IdentityState` will likely be removed with #508 anyway.

Reply:
As a compromise, I would be perfectly fine with making such fields pub(crate), but not pub. With the latter, everything is even more likely to be a breaking change, and we retain less control of the code. Also note that IdentityState will likely be removed with #508 anyway.

Reply:
If one wants to mutate a field defined in the same crate I would recommend declaring the field `pub(crate)` rather than introducing a setter.  For fields one might want to mutate from outside of the crate in which they are defined it can be okay to define a setter if one thinks the internal representation might change in the future, but I would argue for another name in that case, because if the internal representation does in fact change then `set_field_name` becomes a bit misleading. 

Regardless of our preferences, with the `mem::swap` change I suggested above I don't think this method is used (by us) anymore (or am I wrong about that?).